lib/pangea/controllers/class_controller.dart

```diff
@@ -142,6 +143,22 @@ class ClassController extends BaseController {
         );
       }
 
+      // If the room is full, leave
+      final room =
+          _pangeaController.matrixState.client.getRoomById(classChunk.roomId);
+      if (room == null) {
+        return;
+      }
+      if ((await room.leaveIfFull())) {
+        ClassCodeUtil.messageSnack(
```

Comment:
Can you replace the scaffold messages with showFutureLoadingDialogs with errors? Same with the ones in base_analytics.dart. Users might miss these since they show up at the bottom of the screen, rather than in the middle.